### ISSUE TITLE
Remove / Delete module assist (Code action)

### ISSUE THREAD

**Issue by PPakalns** [rust-lang/rust-analyzer#14858]
When refactoring code sometimes some modules becomes empty.

It would be nice if removal of module (recursively) files would be possible.

Why?
Because when you are focused on refactoring code in code editor, switching to file browser to remove files gets distracting. 

Two ideas how this could be implemented:

1) Code action on module definition to remove module and all module files recursively. (This could be bad if files are accidentally deleted)
Maybe it needs to be checked if module has something that it still exports, if not, suggest deletion of module.

2) Show warnings about rust files that are in filesystem, but aren't included in the project. (This could not work in cases where some modules / files are conditionally added to project (cfg definitions based on features))

**Comment by ChayimFriedman2** [comment]
Personally I don't think it deserves an assist. @rust-lang/rust-analyzer?

**Comment by lnicola** [comment]
No, it feels to iffy wrt. files that are not included in the module tree.
